h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -870,7 +870,7 @@ protected ResultInterface queryWithoutCache(int maxRows, ResultTarget target) {
     }
 
     private void disableLazyForJoinSubqueries(TableFilter f) {
-        for (; f != null; f = f.getJoin()) {
+        for (; f != null; f = f.getJoin() != null ? f.getJoin() : f.getNestedJoin()) {
```

Comment:
`TableFilter` can have both `join` and `nestedJoin` at the same time.

Reply:
It's better to use a `TableFilter.visit(TableFilterVisitor)` here.

Reply:
Thanks, fixed.

Reply:
Please, replace your current custom recursive code with `TableFilterVisitor` implementation.

Reply:
Thank for the hint. Fixed.